gtsam/discrete/DecisionTree.h

```diff
@@ -299,6 +299,42 @@ namespace gtsam {
     /// Return the number of leaves in the tree.
     size_t nrLeaves() const;
 
+    /**
+     * @brief This is a convenience function which returns the total number of
```

Comment:
Spelling. And why are we adding it ? And why is the implementation recursive.
I would just as well delete it unless it has a purpose.

Reply:
I'm a bit lost on which word is misspelled. The purpose is to help with testing and ensure correctness as a convenience method.